### PR TITLE
perf(core): use explicit arguments in "eventLoopTick"

### PR DESCRIPTION
**Pull request by bartlomieju**
This commit changes "Deno.core.eventLoopTick" to explicitly accept
two arguments: "opResponses" and "drainNextTick" instead of relying
on implicit "arguments".

Local benchmarks show about 1.5k RPS bump in the `Deno.serve()`
benchmark, but I want to test it on `main` branch.

Ref https://github.com/denoland/deno/pull/18652#discussion_r1165627022

### CODE REVIEW

File: core/01_core.js [hunk 1]
```diff
@@ -148,16 +148,16 @@
...
   // before last are alternating integers and any values that describe the
   // responses of async ops.
-  function eventLoopTick() {
+  function eventLoopTick(opResponses, drainNextTick) {
     // First respond to all pending ops.
-    for (let i = 0; i < arguments.length - 1; i += 2) {
-      const promiseId = arguments[i];
-      const res = arguments[i + 1];
+    for (let i = 0; i < opResponses.length - 1; i += 2) {
+      const promiseId = opResponses[i];
```
Comment (mmastrac) [inline core/01_core.js:154]: If we split `opResponses` into two arrays, the `promiseId` one can be a `UInt32Array` and may perform slightly better.

**Review by littledivy — COMMENTED** [review]
Are you sure this is better then using spread for arguments? This is creating an Array for every tick while arguments passed are already in an array.

**Comment by bartlomieju** [comment]
> Are you sure this is better then using spread for arguments? This is creating an Array for every tick while arguments passed are already in an array.

I'm not sure I understand - the array was created before this PR as well - I just moved that array to be the first argument instead of relying on implicit `arguments` arg. Using spread operator shows up on flamegraphs for `opAsync()` so I'm pretty sure it will show up here as well.

I will try Matt's suggestion about using two separate arrays before landing this one.

**Comment by bartlomieju** [comment]
`main` branch:
```
time 1658 ms rate 603136
time 1627 ms rate 614628
time 1638 ms rate 610500
time 1613 ms rate 619962
time 1613 ms rate 619962
time 1616 ms rate 618811
time 1621 ms rate 616903
time 1622 ms rate 616522
time 1611 ms rate 620732
time 1606 ms rate 622665
time 1598 ms rate 625782
time 1602 ms rate 624219
time 1597 ms rate 626174
time 1597 ms rate 626174
time 1615 ms rate 619195
time 1620 ms rate 617283
time 1592 ms rate 628140
time 1601 ms rate 624609
```

This PR:
```
time 1698 ms rate 588928
time 1681 ms rate 594883
time 1662 ms rate 601684
time 1672 ms rate 598086
time 1677 ms rate 596302
time 1674 ms rate 597371
time 1684 ms rate 593824
time 1657 ms rate 603500
time 1656 ms rate 603864
time 1669 ms rate 599161
```

Appears this one is actually slower.

**Comment by bartlomieju** [comment]
No perf gain, closing for now.

**Pull request closed without merging**